lib/common/exceptions/external/backend_service_exception.rb

```diff
@@ -1,25 +1,96 @@
+      # not the I18n version.
+      def message
+        "BackendServiceException: #{response_values.merge(code: code)}"
       end
 
       def errors
-        Array(SerializableError.new(i18n_data.merge(detail: @detail, source: @source, meta: @meta)))
+        Array(SerializableError.new(i18n_data.merge(render_overides)))
       end
 
       private
 
+      def render_overides
+        { status: status, detail: detail, code: code, source: source }
+      end
+
+      # REQUIRED - This is the i18n code returned from raise_error middleware. If it exists in
+      # I18n then it should be like RX139 or EVSS144, otherwise VA900
+      def code
+        if @key.present? && I18n.exists?("common.exceptions.#{@key}")
+          @key
+        else
+          'VA900'
+        end
+      end
+
+      # REQUIRED - This is the http status code.
+      # unless you've specified that you want the status code to be something other
+      # then 400 explicitly it will default to 400. IT WILL NOT DEFAULT to whatever
+      # was provided by the backend service, because the backend service response
+      # might not always be relevant
+      def status
+        i18n_data[:source].presence || 400
+      end
+
+      # OPTIONAL - This is the detail or message that is rendered in JSON response
+      # Not providing detail will render a detail the same as title, 'Operation failed'
+      # NOTE: in the future, detail will only work via i18n, not the value from response_values
+      def detail
+        i18n_data[:detail].presence || response_values[:detail]
+      end
+
+      # OPTIONAL - This should usually be a developer message of some sort from the backend service
+      # if one is not provided by the backend this can be nil and the key will not be rendered
+      def source
+        response_values[:source]
+      end
+
+      def validate_arguments!
+        raise ArgumentError, "i18n key (#{@key}) is invalid" unless I18n.exists?(i18n_key)
+        raise ArgumentError, "status (#{status}) is not in range" unless status.between?(400, 599)
+      end
+
+      # This just reports to Sentry that an unmapped backend service exception was
+      # identified it does not actually raise an exception.
+      # NOTE: in the future detail will just fallback to 'Operation failed'
+      def warn_about_error_not_in_locales!
+        unless i18n_data[:detail].present?
+          message = <<-MESSAGE.strip_heredoc
+            Referencing detail from response values is deprecated. Add the following to exceptions.en.yml
+            #{response_values[:code]}:
+              code: '#{response_values[:code]}'
+              detail: '#{response_values[:detail]}'
+              status: <http status code you want rendered (400 or 422)>
+              source: ~
+          MESSAGE
+          Rails.logger.warn message
+          exception = UnmappedBackendServiceException.new(message)
+          Raven.capture_exception(exception) if ENV['SENTRY_DSN'].present?
```

Comment:
Could this use `capture_message` instead?
https://docs.sentry.io/clients/ruby/usage/#reporting-messages